lib/elixir/lib/kernel.ex

```diff
@@ -1331,7 +1331,7 @@ defmodule Kernel do
   If the `right` operand is not a proper list, it returns an improper list.
   If the `left` operand is not a proper list, it raises `ArgumentError`.
 
-  Inlined by the compiler.
+  Not allowed in guard clauses. Inlined by the compiler.
```

Comment:
We never mention when it is not allowed. Only when it is

Reply:
I thought it was being used because of these:

https://github.com/elixir-lang/elixir/blob/75726d5611413ee45cb5235b1698944b72efa244/lib/elixir/lib/kernel.ex#L1743

https://github.com/elixir-lang/elixir/blob/75726d5611413ee45cb5235b1698944b72efa244/lib/elixir/lib/kernel.ex#L3496

https://github.com/elixir-lang/elixir/blob/75726d5611413ee45cb5235b1698944b72efa244/lib/elixir/lib/kernel.ex#L3536

I can remove the documentation changes if they are needed though.

Reply:
That is because the strickly boolean operator counter part is allowed.

Reply:
I reverted the change that added the doc changes, thanks!